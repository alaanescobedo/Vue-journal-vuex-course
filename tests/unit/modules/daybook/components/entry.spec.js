import EntryComponent from '@/modules/daybook/components/Entry';
import { shallowMount } from '@vue/test-utils';
import { journalState } from '../../../mock/test-journal-state';
import getDayMonthYear from '@/modules/daybook/helpers/getDayMonthYear';

describe('Entry Component', () => {
  const mockRouter = {
    push: jest.fn(),
  };
  const mockEntry = journalState.entries[0];

  const wrapper = shallowMount(EntryComponent, {
    props: {
      entry: mockEntry,
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  test('debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('debe de redireccionar al hacer click en el container', () => {
    const entryContainer = wrapper.find('.entry-container');
    entryContainer.trigger('click');

    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'entry', params: { id: mockEntry.id } });
  });
  test('pruebas en las propiedades computadas', () => {
    const { day, month, year } = getDayMonthYear(mockEntry.date);

    expect(wrapper.vm.day).toBe(day);
    expect(wrapper.vm.month).toBe(month);
    expect(wrapper.vm.year).toBe(year);
  });
});
