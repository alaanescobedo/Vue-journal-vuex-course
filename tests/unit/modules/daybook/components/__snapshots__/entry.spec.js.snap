// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Component debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">25</span><span class="mx-1 fs-5">Agosto</span><span class="mx-2 fs-5 fw-light">2021, Mi√©rcoles</span></div>
  <div class="entry-description">Angular</div>
</div>
`;
